From 8a287d2f54e83596d6535cb45d7933b2dd72068e Mon Sep 17 00:00:00 2001
From: Dinesh Sudham <dineshx.sudham@intel.com>
Date: Wed, 8 Jul 2020 15:37:07 +0800
Subject: [PATCH] DRVLIB_SW-3518:GSWAPI: Setting default portmap for external
 switch

This controls the handling of ethernet frames
that have a target portmap

Signed-off-by: Dinesh Sudham <dineshx.sudham@intel.com>
---
 drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c b/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c
index 30c4cd314e77..65c3bbdf8a01 100644
--- a/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c
+++ b/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c
@@ -2991,6 +2991,15 @@ static void prx300_gsw_ext_config(void *cdev)
 		gsw_w32(cdev, (PCE_PCTRL_3_LNDIS_OFFSET + (0xA * port_idx)),
 			PCE_PCTRL_3_LNDIS_SHIFT,
 			PCE_PCTRL_3_LNDIS_SIZE, 1);
+	   /* Use default port Forwarding */
+	gsw_w32(cdev, PCE_GCTRL_0_NOPM_OFFSET,
+		PCE_GCTRL_0_NOPM_SHIFT,
+		PCE_GCTRL_0_NOPM_SIZE, 1);
+	/* VLAN aware switching */
+	gsw_w32(cdev, PCE_GCTRL_0_VLAN_OFFSET,
+		PCE_GCTRL_0_VLAN_SHIFT,
+		PCE_GCTRL_0_VLAN_SIZE,
+		1);
 
 	gsw_w32(cdev, SGMII_TBI_LPSTAT_OFFSET,
 		SGMII_TBI_LPSTAT_SHIFT,
